src/gui/editors/AutomationEditor.cpp

```diff
@@ -1709,6 +1718,61 @@ void AutomationEditor::wheelEvent(QWheelEvent * we )
 	{
 		m_topBottomScroll->setValue( m_topBottomScroll->value() -
 							we->delta() / 30 );
+		
+		if (we->y() > TOP_MARGIN) {
+			float level = getLevel( we->y() );
+			int x = we->x();
+
+			if (x >= VALUES_WIDTH) {
+		// set or move value
+
+				x -= VALUES_WIDTH;
+			// get tick in which the cursor is posated
+				int pos_ticks = x * MidiTime::ticksPerTact() / m_ppt +
+								m_currentPosition;
+
+				// get time map of current pattern
+				timeMap & time_map = m_pattern->getTimeMap();
+
+				// will be our iterator in the following loop
+				timeMap::iterator it = time_map.begin();
+
+				// and check whether the user scrolls over an
+				// existing value
+				while (it != time_map.end()) {
+					
+					if (pos_ticks < 0) {
+						pos_ticks = 0;
+					}
+							
+					if (pos_ticks >= it.key() - MidiTime::ticksPerTact() *4 / m_ppt
+						&& (it+1==time_map.end() ||	pos_ticks <= (it+1).key())
+						&& (pos_ticks<= it.key() + MidiTime::ticksPerTact() *4 / m_ppt)) 
+					{
+						// mouse wheel up
+						if (we->delta() < 0) {
+							level = roundf(it.value() * 1000) / 1000 -
+							 m_pattern->firstObject()->step<float>();
+						// mouse wheel down
+						} else if (we->delta() > 0)	{
+							level = roundf(it.value() * 1000) / 1000 +
+							 m_pattern->firstObject()->step<float>();
```

Comment:
Sure.

Here again, I was trying to follow conventions and wasn't sure.  Spaces before and after operators.  I suppose, however, the space before the line is more than enough.

Reply:
Yes spaces should be placed before and after operators thats good, but this line is split. 

I'm sorry, made a mistake here. Can you instead of removing the space, replace it with one tab? So split lines are indented. (since we use tabs as indentation) It still fits the max 80 chars per line then.

```cpp
							level = roundf(it.value() * 1000) / 1000 +
								m_pattern->firstObject()->step<float>();
```

Reply:
No problem.